gtsam/discrete/Assignment.h

```diff
@@ -58,27 +58,29 @@ namespace gtsam {
    * variables with each having cardinalities 4, we get 4096 possible
    * configurations!!
    */
-  template<typename L>
-  std::vector<Assignment<L> > cartesianProduct(
-      const std::vector<std::pair<L, size_t> >& keys) {
-    std::vector<Assignment<L> > allPossValues;
+  template <typename L>
+  std::vector<Assignment<L>> cartesianProduct(const std::map<L, size_t>& keys) {
```

Comment:
Oopsies, Nope! CartesianProduct very deliberately takes vectors! The order in which things are enumerated depends on it.